torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -61,6 +64,11 @@ Reducer::Reducer(
       AT_ASSERTM(
           replicas_[replica_index].size() == replicas_[0].size(),
           "Model replicas must have an equal number of parameters.");
+      AT_ASSERTM(
```

Comment:
Q: (not necessary to make any changes here) as `expect_sparse_gradients_ ` should be the same for all replicas, do we still need a 2D list there? (same for `bucket_indices`)

Reply:
It should be the same, indeed. I rather have the additional check here than deal with runtime errors where they are not the same. At least with this in place we can make those checks and it will barf if there is any asymmetry between replicas.